### PR TITLE
DOC warn user about potential information loss in Resampler.interpolate

### PR DESCRIPTION
**Pull request by kopytjuk**
In scientific and technical domain people deal with high-frequent or non-equidistant timeseries. Using `resample("1s").interpolate()` can have unwanted side effects which we should warn in the documentation:

```python
import datetime as dt
import pandas as pd
import matplotlib.pyplot as plt

timesteps = [
    dt.datetime(2023, 3, 1, 7, 0, 0),
    dt.datetime(2023, 3, 1, 7, 0, 1),
    dt.datetime(2023, 3, 1, 7, 0, 2),
    dt.datetime(2023, 3, 1, 7, 0, 3),
    dt.datetime(2023, 3, 1, 7, 0, 4)
]
series = pd.Series(data=[1, -1, 2, 1, 3], index=timesteps)

resample_freq = "400ms"
series_resampler = series.resample(resample_freq)
series_resampled_linear = series_resampler.interpolate("linear")
```
which leads to the following:

result

For that example, information loss is expected.

Imho we should warn the user about this behavior. It seems to be known, to quote @jreback from [here](https://github.com/pandas-dev/pandas/issues/14297#issuecomment-285434729):

> so the reason this happens is because the index is first reindexed to the new time buckets (upsampled) via reindexing, then interpolation happens.  

- [x] refers to #12552, #14297
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/resample.py [hunk 1]
```diff
@@ -839,7 +839,24 @@ def interpolate(
...
     ):
         """
-        Interpolate values according to different methods.
+        Interpolate values according to different methods. Note that the points used
+        as anchor points in the interpolation are taken from the original timeseries
+        and thus it can lead to information loss and wrong interpolation!
+        
+        ```
+        # example of 2Hz timeseries resampled with rule="400ms"
+        2023-03-01 07:00:00.000    1.0  <- anchor 0
```
Comment (mroeschke) [inline pandas/core/resample.py:848]: This would be better shown in the `Examples` section of the docstring

Comment (kopytjuk) [reply pandas/core/resample.py:896]: Good suggestion - I introduced that.

**Comment by mroeschke** [comment]
Thanks, looks like some of our docstring validations are failing

File: pandas/core/resample.py [hunk 2]
```diff
@@ -839,8 +839,74 @@ def interpolate(
...
+        2023-03-01 07:00:01.500    0.5
+        2023-03-01 07:00:02.000    2.0
+        2023-03-01 07:00:02.500    1.5
+        2023-03-01 07:00:03.000    1.0
+        2023-03-01 07:00:03.500    2.0
+        2023-03-01 07:00:04.000    3.0
+        Freq: 500L, dtype: float64
+
+        Internal reindexing prior to interpolation leads to a timeseries
+        interpolated given the reindexed timestamps (anchros) which can lead
```
Comment (mroeschke) [inline pandas/core/resample.py:892]:
```suggestion
        interpolated given the reindexed timestamps (anchors) which can lead
```

File: pandas/core/resample.py [hunk 3]
```diff
@@ -839,8 +839,74 @@ def interpolate(
...
+        to misleading interpolation results:
+
+        >>> series.resample("400ms").interpolate("linear")
+        2023-03-01 07:00:00.000    1.0  <- anchor 0
```
Comment (mroeschke) [inline pandas/core/resample.py:896]: You annotations of `anchor` will probably need to be described in the paragraph above The doctests checks the exact output

File: pandas/core/resample.py [hunk 4]
```diff
@@ -825,7 +825,6 @@ def fillna(self, method, limit: int | None = None):
         """
         return self._upsample(method, limit=limit)
 
-    @doc(NDFrame.interpolate, **_shared_docs_kwargs)
```
Comment (kopytjuk) [inline pandas/core/resample.py:828]: @mroeschke here I remove the docstring concatenation and added links to documentation, so the user can inform her/himself about various methods

**Comment by kopytjuk** [comment]
@mroeschke let me know if something is missing/to be extended

File: pandas/core/resample.py [hunk 5]
```diff
@@ -840,7 +839,90 @@ def interpolate(
...
+        2023-03-01 07:00:02.800    2.4
+        2023-03-01 07:00:03.200    2.6
+        2023-03-01 07:00:03.600    2.8
+        2023-03-01 07:00:04.000    3.0
+        Freq: 400L, dtype: float64
+
+        Note that the series erroneously increases between two anchors
+        ``07:00:00`` and ``07:00:02``.
         """
+
```
Comment (mroeschke) [inline pandas/core/resample.py:925]: Can you undo this change?

File: pandas/core/resample.py [hunk 6]
```diff
@@ -840,7 +839,90 @@ def interpolate(
     ):
         """
         Interpolate values according to different methods.
+
```
Comment (mroeschke) [inline pandas/core/resample.py:842]: Can you document all the parameters too?

File: pandas/core/resample.py (hunk 5 above)
Comment (kopytjuk) [reply pandas/core/resample.py]: Removed.

File: pandas/core/resample.py (hunk 6 above)
Comment (kopytjuk) [reply pandas/core/resample.py:915]: Done.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @kopytjuk
